src/core/connection.c

```diff
@@ -4823,6 +4823,20 @@ QuicConnRecvFrames(
                 break; // Ignore frame if we are closed.
             }
 
+            uint8_t Type =
+                (QuicConnIsServer(Connection) ? // Peer's role, so flip
+                STREAM_ID_FLAG_IS_CLIENT : STREAM_ID_FLAG_IS_SERVER)
+                |
+                (Frame.BidirectionalStreams ?
+                 STREAM_ID_FLAG_IS_BI_DIR : STREAM_ID_FLAG_IS_UNI_DIR);
+
+            const QUIC_STREAM_TYPE_INFO* Info = &Connection->Streams.Types[Type];
+
+            if (Info->MaxTotalStreamCount > Frame.StreamLimit)
+            {
+                break;
+            }
```

Comment:
1. Put this after marking the packet for ACK.
2. Fix the `if` so that the `{` is on the same line.